Sources/PortalSwift/Components/PortalWebView.swift

```diff
@@ -147,6 +141,54 @@ public class PortalWebView: UIViewController, WKNavigationDelegate, WKScriptMess
       self.webViewContentIsLoaded = true
     }
   }
+  
+  private func initWebView(address: String, persistSessionData: Bool = false) -> WKWebView {
+    return {
+      // build WKUserScript
+      let scriptSource = self.injectPortal(
+        address: address,
+        apiKey: self.portal.apiKey,
+        chainId: String(self.portal.chainId),
+        gatewayConfig: self.portal.gatewayConfig[self.portal.chainId]!,
+        autoApprove: self.portal.autoApprove,
+        enableMpc: true
+      )
+      let script = WKUserScript(source: scriptSource, injectionTime: .atDocumentStart, forMainFrameOnly: true)
+      
+      // build the WekUserContentController
+      let contentController = WKUserContentController()
+      contentController.addUserScript(script)
+      contentController.add(self, name: "WebViewControllerMessageHandler")
+
+      // build the WKWebViewConfiguration
+      let configuration = WKWebViewConfiguration()
+      configuration.userContentController = contentController
+
+      // Allows for data persistence across sessions
+      if persistSessionData {
+       configuration.websiteDataStore = WKWebsiteDataStore.default()
+      }
+
+      let webView = WKWebView(frame: .zero, configuration: configuration)
+      webView.scrollView.bounces = false
+      webView.navigationDelegate = self
+      webView.uiDelegate = self
+
+
+      // Enable debugging the webview in Safari.
```

Comment:
maybe we can add a debug flag to the args for the init function. Im sure customers would want to use it as well

Reply:
Its currently using compiler directives to determine whether to enable that flag. When we package the code, is it cool to leave compiler directives in?